src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -883,6 +883,60 @@ class npc_ros_dark_rider : public CreatureScript
 
 };
 
+enum GiftOfTheHarvester
+{
+    SPELL_GIFT_OF_THE_HARVESTER = 52479,
+    SPELL_MINER_GHOUL_TRANSFORM = 52490,
+    SPELL_MINER_GHOST_TRANSFORM = 52505
+};
+
+struct gift_of_the_harvesterAI : public GameObjectAI
+{
+    gift_of_the_harvesterAI(GameObject* go) : GameObjectAI(go) { }
+
+    void Reset() override
+    {
+        _scheduler.Schedule(2s, [this](TaskContext /*context*/)
+        {
+            me->SendCustomAnim(0);
+            me->CastSpell(nullptr, SPELL_GIFT_OF_THE_HARVESTER, true);
+        });
+    }
+    
+    void UpdateAI(uint32 diff) override
+    {
+        _scheduler.Update(diff);
+    }
+
+private:
+    TaskScheduler _scheduler;
+};
+
+struct spell_gift_of_the_harvester : public SpellScript
+{
+    PrepareSpellScript(spell_gift_of_the_harvester);
+
+    bool Validate(SpellInfo const* /*spell*/) override
+    {
+        return ValidateSpellInfo(
+                {
+                    SPELL_MINER_GHOUL_TRANSFORM,
+                    SPELL_MINER_GHOST_TRANSFORM
+                });
+    }
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        uint32 spellId = urand(0, 1) ? SPELL_MINER_GHOUL_TRANSFORM : SPELL_MINER_GHOST_TRANSFORM;
```

Comment:
use the random uint32 picker, not urand

Reply:
RAND

Reply:
btw why the hell is RAND inside CreatureAIImpl.h ??

Reply:
You mean this https://github.com/TrinityCore/TrinityCore/pull/23114/commits/fd64cb28547c998a6e8700c8590614b1b5cb6a4f?

Reply:
No pratical change ._.

Reply:
yep
@Keader the concept is: if there is a helper designed to do something, use it